packages/integration-karma/test/rendering/iteration/index.spec.js

```diff
@@ -0,0 +1,59 @@
+import { extractDataIds } from 'test-utils';
+import { createElement } from 'lwc';
+import ForEach from 'x/forEach';
+import Iterator from 'x/iterator';
+import InlineForEach from 'x/inlineForEach';
+import InlineIterator from 'x/inlineIterator';
+
+describe('iteration rendering', () => {
+    function validateRenderedChildren(elm, iterationType) {
+        const isIterator = iterationType === 'iterator';
+        const expectedChildren = elm.items;
+        const { children } = elm.shadowRoot;
+        expect(children.length).toEqual(expectedChildren.length);
+
+        for (let i = 0; i < children.length; i++) {
+            const child = children[i];
+            const expected = expectedChildren[i];
+            const dataIds = extractDataIds(child);
+            expect(dataIds.first.textContent).toEqual(isIterator ? (i === 0).toString() : '');
+            expect(dataIds.last.textContent).toEqual(
+                isIterator ? (i === children.length - 1).toString() : ''
+            );
+            expect(dataIds.index.textContent).toEqual(i.toString());
+            expect(dataIds.value.textContent).toEqual(expected.toString());
+        }
+    }
+
+    const components = {
+        'x-inline-for-each': InlineForEach,
+        'x-inline-iterator': InlineIterator,
+        'x-template-for-each': ForEach,
+        'x-template-iterator': Iterator,
+    };
+
+    ['inline'].forEach((type) => {
+        ['iterator'].forEach((iterationType) => {
```

Comment:
```suggestion
    ['template', 'inline'].forEach((type) => {
        ['for-each', 'iterator'].forEach((iterationType) => {
```

Reply:
Why not use `Object.entries` to avoid the nested loop?

Reply:
Because I wasn't thinking too much about it. :) I can change it.

And yes, this was a typo. I'll switch to Object.entries.